Reporting/M5 Document/Reflection.md

```diff
@@ -0,0 +1,29 @@
+## How did your project management work for the team?  What was the hardest thing, and what would you do the same/differently the next time you plan to complete a project like this? 
+
+- Our project management worked relatively well for the team. We maintained regular communication channels, utilized project management tools effectively, and delegated tasks efficiently. However, the hardest aspect was ensuring effective coordination between front-end and back-end development teams, especially when integrating complex features like peer-to-peer messaging and encryption. 
+
+- In future projects, we would prioritize more frequent cross-team check-ins and perhaps consider adopting a more integrated development approach from the outset to mitigate such challenges.
+
+## Do you feel that your initial requirements were sufficiently detailed for this project?  Which requirements did you miss or overlook?
+- While our initial requirements provided a solid foundation for the project, there were some areas where additional detail could have been beneficial. Specifically, we overlooked the specifics of password security measures, and multimedia messaging capabilities. 
+
+- In the future, we would ensure a more comprehensive review of security and messaging features during the requirement gathering phase to avoid oversights.
+
+## What did you miss in your initial planning for the project (beyond just the requirements)?
+- In our initial planning, we may have overlooked the potential impact of external dependencies and integration challenges. Specifically, we did not anticipate the complexities associated with integrating encryption protocols into our messaging system or the time required for thorough testing across different platforms and devices.
```

Comment:
Encryption was never a requirement of our project (user stories etc.), but it was mentioned in the project description / introduction. What we missed in our initial planning was planning out the system design of our software in depth and also revising it, as for example, we had overlooked the idea of authentication and web sockets to facilitate accounts and real-time operations like synchronizing mutual refresh operations across users in real-time.